src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/QueueHealthDecider.java

```diff
@@ -121,10 +134,8 @@ private ModifyQueueCapacityAction configureQueueCapacity(NodeKey esNode, Resourc
   }
 
   private int getNodeQueueCapacity(NodeKey esNode, ResourceEnum threadPool) {
-    // TODO: use NodeConfigurationRca to return capacity, for now returning defaults
-    if (threadPool.equals(ResourceEnum.SEARCH_THREADPOOL)) {
-      return 1000;
-    }
-    return 100;
+      return (int) getAppContext().getNodeConfigCache().get(esNode, Resource.newBuilder()
```

Comment:
This will return an `IllegalArgumentException` if the NodeKey is not present. We should add handling for it.

Reply:
I have handled exceptions in the caller of this function. It will result in no action being taken on the cluster.